cmd/dep/root_analyzer.go

```diff
@@ -150,19 +150,37 @@ func (a *rootAnalyzer) Info() (string, int) {
 	return name, version
 }
 
-func lookupVersionForRevision(rev gps.Revision, pi gps.ProjectIdentifier, sm gps.SourceManager) (gps.Version, error) {
+// lookupVersionForLockedProject figures out the appropriate version for a locked
+// project based on the locked revision and the constraint from the manifest.
+// First try matching the revision to a version, then try the constraint from the
+// manifest, then finally the revision.
+func lookupVersionForLockedProject(pi gps.ProjectIdentifier, c gps.Constraint, rev gps.Revision, sm gps.SourceManager) (version gps.Version, warning error) {
```

Comment:
What's the benefit of named return values here?

Also, naming the `error` `warning` seems a bit misleading.

For `lookupVersionForLockedProject`, it's an error that prevents returning the expected results. Consumers cannot rely on the version returned since it will be `nil`.

A consumer can call this value a `warning` if it doesn't depend on getting a valid `gps.Version` value. But I would only call it `warning` here if consumers could, somehow, use the returned value if they didn't care about the warning.